CI: TestNodeManagerEmitStatus : panic: test timed out after

## CI failure: TestNodeManagerEmitStatus

```
PANIC  package: github.com/cilium/cilium/pkg/node/manager • TestNodeManagerEmitStatus
panic: test timed out after 12m0s                                                                                                                                                     
	running tests:                                                                                                                                                                        
		TestNodeManagerEmitStatus (11m59s)                                                                                                                                                    
                                                                                                                                                                                      
goroutine 42 [running]:                                                                                                                                                               
testing.(*M).startAlarm.func1()                                                                                                                                                       
	/root/sdk/go1.24.0/src/testing/testing.go:2484 +0x394                                                                                                                                 
created by time.goFunc                                                                                                                                                                
	/root/sdk/go1.24.0/src/time/sleep.go:215 +0x2d                                                                                                                                        
                                                                                                                                                                                      
goroutine 1 [chan receive, 11 minutes]:                                                                                                                                               
testing.(*T).Run(0xc000103a40, {0x33f0669?, 0xc000691b30?}, 0x35236e8)                                                                                                                
	/root/sdk/go1.24.0/src/testing/testing.go:1859 +0x431                                                                                                                                 
testing.runTests.func1(0xc000103a40)                                                                                                                                                  
	/root/sdk/go1.24.0/src/testing/testing.go:2279 +0x37                                                                                                                                  
testing.tRunner(0xc000103a40, 0xc000691c70)                                                                                                                                           
	/root/sdk/go1.24.0/src/testing/testing.go:1792 +0xf4                                                                                                                                  
testing.runTests(0xc000689f20, {0x5861ee0, 0x11, 0x11}, {0x5889300?, 0x7?, 0x5886da0?})                                                                                               
	/root/sdk/go1.24.0/src/testing/testing.go:2277 +0x4b4                                                                                                                                 
testing.(*M).Run(0xc000882c80)                                                                                                                                                        
	/root/sdk/go1.24.0/src/testing/testing.go:2142 +0x64a                                                                                                                                 
main.main()                                                                                                                                                                           
	_testmain.go:89 +0x9b                                                                                                                                                                 
                                                                                                                                                                                      
goroutine 35 [chan receive, 11 minutes]:                                                                                                                                              
github.com/cilium/cilium/pkg/node/manager.TestNodeManagerEmitStatus(0xc00057ac40)                                                                                                     
	/host/pkg/node/manager/manager_test.go:953 +0xb16                                                                                                                                     
testing.tRunner(0xc00057ac40, 0x35236e8)                                                                                                                                              
	/root/sdk/go1.24.0/src/testing/testing.go:1792 +0xf4                                                                                                                                  
created by testing.(*T).Run in goroutine 1                                                                                                                                            
	/root/sdk/go1.24.0/src/testing/testing.go:1851 +0x413                                                                                                                                 
                                                                                                                                                                                      
goroutine 37 [select, 11 minutes]:                                                                                                                                                    
github.com/cilium/statedb.graveyardWorker(0xc000a18ee8, {0x381b4a8, 0xc00002a5a0}, 0x3b9aca00?)                                                                                       
	/host/vendor/github.com/cilium/statedb/graveyard.go:25 +0x17a                                                                                                                         
created by github.com/cilium/statedb.(*DB).Start in goroutine 35                                                                                                                      
	/host/vendor/github.com/cilium/statedb/db.go:281 +0x165                                                                                                                               
                                                                                                                                                                                      
goroutine 38 [select]:                                                                                                                                                                
github.com/cilium/cilium/pkg/trigger.(*Trigger).waiter(0xc000883720)                                                                                                                  
	/host/pkg/trigger/trigger.go:209 +0x285                                                                                                                                               
created by github.com/cilium/cilium/pkg/trigger.NewTrigger in goroutine 35                                                                                                            
	/host/pkg/trigger/trigger.go:121 +0x1d8                                                                                                                                               
                                                                                                                                                                                      
goroutine 39 [chan send, 11 minutes]:                                                                                                                                                 
github.com/cilium/cilium/pkg/node/manager.(*signalNodeHandler).NodeValidateImplementation(_, {{0x33b76c8, 0x5}, {0x33b51bc, 0x2}, {0x0, 0x0, 0x0}, 0x0, {0x0, ...}, ...})             
	/host/pkg/node/manager/manager_test.go:220 +0x2a                                                                                                                                      
github.com/cilium/cilium/pkg/node/manager.(*manager).singleBackgroundLoop.func1({0x3826bd0, 0xc00119b180})                                                                            
	/host/pkg/node/manager/manager.go:584 +0x9a                                                                                                                                           
github.com/cilium/cilium/pkg/node/manager.(*manager).Iter(0x2e92ee0?, 0xc000241978)                                                                                                   
	/host/pkg/node/manager/manager.go:226 +0x102                                                                                                                                          
github.com/cilium/cilium/pkg/node/manager.(*manager).singleBackgroundLoop(0xc0011b0900, {0x381b470, 0xc0011f7350}, 0xa939b)                                                           
	/host/pkg/node/manager/manager.go:583 +0x490                                                                                                                                          
github.com/cilium/cilium/pkg/node/manager.(*manager).backgroundSync(0xc0011b0900, {0x381b470, 0xc0011f7350}, {0x381fd20, 0xc000381680})                                               
	/host/pkg/node/manager/manager.go:508 +0x26d                                                                                                                                          
github.com/cilium/hive/job.(*jobOneShot).start(0xc00004e900, {0x381b470, 0xc0011f7350}, 0x0?, {0x381fd20, 0xc00004e660}, {{{0xc0011a0560, 0x1, 0x1}}, 0xc00031f6a0, ...})             
	/host/vendor/github.com/cilium/hive/job/oneshot.go:138 +0x4fd                                                                                                                         
created by github.com/cilium/hive/job.(*group).add.func1 in goroutine 35                                                                                                              
	/host/vendor/github.com/cilium/hive/job/job.go:209 +0x172                                                                                                                             
                                                                                                                                                                                      
goroutine 40 [select, 11 minutes]:                                                                                                                                                    
github.com/cilium/cilium/pkg/node/manager.(*manager).carrierDownReconciler(0xc0011b0900, {0x381b470, 0xc0011f7380}, {0x381fd20, 0xc000380c00})                                        
	/host/pkg/node/manager/manager.go:427 +0x34c                                                                                                                                          
github.com/cilium/hive/job.(*jobOneShot).start(0xc00004e960, {0x381b470, 0xc0011f7380}, 0x0?, {0x381fd20, 0xc00004e660}, {{{0xc0011a0560, 0x1, 0x1}}, 0xc00031f6a0, ...})             
	/host/vendor/github.com/cilium/hive/job/oneshot.go:138 +0x4fd                                                                                                                         
created by github.com/cilium/hive/job.(*group).add.func1 in goroutine 35                                                                                                              
	/host/vendor/github.com/cilium/hive/job/job.go:209 +0x172                                                                                                                             
                                                                                                                                                                                      
goroutine 41 [select, 11 minutes]:                                                                                                                                                    
github.com/cilium/cilium/pkg/hive/health.publishJob({0x381b470, 0xc0011f7500}, {{{}}, 0xc000a19110, {0x3869240, 0xc00119b000}, {0x381b6e0, 0xc0004a1cb0}, 0xc0011a0c20}, 0xc000a38ec0)
	/host/pkg/hive/health/metrics.go:120 +0x34f                                                                                                                                           
github.com/cilium/cilium/pkg/hive/health.metricPublisher.func2({0x381b470?, 0xc0011f7500?}, {0x7?, 0xfffffffffffffffc?})                                                              
	/host/pkg/hive/health/metrics.go:79 +0x39                                                                                                                                             
github.com/cilium/hive/job.(*jobOneShot).start(0xc00004e840, {0x381b470, 0xc0011f7500}, 0x0?, {0x381fd20, 0xc00004e720}, {{{0xc0011a0b00, 0x1, 0x1}}, 0xc00031f0c0, ...})             
	/host/vendor/github.com/cilium/hive/job/oneshot.go:138 +0x4fd                                                                                                                         
created by github.com/cilium/hive/job.(*group).Start.func1 in goroutine 35                                                                                                            
	/host/vendor/github.com/cilium/hive/job/job.go:161 +0x17c                                                                                                                             
                                                                                                                                                                                      
goroutine 50 [select]:                                                                                                                                                                
github.com/cilium/cilium/pkg/rate.NewLimiter.func1()                                                                                                                                  
	/host/pkg/rate/limiter.go:47 +0x93                                                                                                                                                    
created by github.com/cilium/cilium/pkg/rate.NewLimiter in goroutine 41                                                                                                               
	/host/pkg/rate/limiter.go:45 +0x127                        
```

https://github.com/cilium/cilium/actions/runs/13417441188/job/37481832373#step:17:17531

Local repro seems easy:

`go test -race ./pkg/node/manager -run "TestNodeManagerEmitStatus" -timeout=10s` fails. So does `stress ./manager.test -test.run "TestNodeManagerEmitStatus" -test.timeout=10s`

Given the test was revamped recently I'll assign @dylandreimerink as the author - feel free to punt if you don't think the changes are to blame.

--- a/pkg/node/manager/manager_test.go
+++ b/pkg/node/manager/manager_test.go
@@ -176,6 +176,7 @@ type signalNodeHandler struct {
 	NodeValidateImplementationEvent       chan nodeTypes.Node
 	NodeValidateImplementationEventError  error
 	EnableNodeValidateImplementationEvent bool
+	Stop                                  chan struct{}
 }
 
 func newSignalNodeHandler() *signalNodeHandler {
@@ -184,6 +185,7 @@ func newSignalNodeHandler() *signalNodeHandler {
 		NodeUpdateEvent:                 make(chan nodeTypes.Node, 10),
 		NodeDeleteEvent:                 make(chan nodeTypes.Node, 10),
 		NodeValidateImplementationEvent: make(chan nodeTypes.Node, 4096),
+		Stop:                            make(chan struct{}, 10),
 	}
 }
 
@@ -217,7 +219,10 @@ func (n *signalNodeHandler) AllNodeValidateImplementation() {
 
 func (n *signalNodeHandler) NodeValidateImplementation(node nodeTypes.Node) error {
 	if n.EnableNodeValidateImplementationEvent {
-		n.NodeValidateImplementationEvent <- node
+		select {
+		case <-n.Stop:
+		case n.NodeValidateImplementationEvent <- node:
+		}
 	}
 	return n.NodeValidateImplementationEventError
 }
@@ -946,11 +951,9 @@ func TestNodeManagerEmitStatus(t *testing.T) {
 	status, _ = checkStatus()
 	assert.Equal(types.LevelOK, string(status.Level))
 
-	// Replace the channel with a buffered one and empty the old channel.
-	// This unblocks the background sync, so everything can exit as it should.
-	oldChan := nh1.NodeValidateImplementationEvent
-	nh1.NodeValidateImplementationEvent = make(chan nodeTypes.Node, 100)
-	<-oldChan
+	for i := 0; i < cap(nh1.Stop); i++ {
+		nh1.Stop <- struct{}{}
+	}
 }
 
 // TestCarrierDownReconciler tests that we can detect carrier down events for physical devices

